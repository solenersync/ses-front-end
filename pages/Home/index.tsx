import { StarIcon } from "@heroicons/react/20/solid";
import Footer from "components/Footer/footer";
import { NextPage } from "next";
import Link from "next/link";
import { useRouter } from "next/router";
import { useState } from "react";

const Home: NextPage = (props) => {
  const router = useRouter();
  const [email, setEmail] = useState("");

  const handleSubmit = (e: any) => {
    e.preventDefault();
    router.push({ pathname: "/signup", query: { email: email } });
  };

  return (
    <div className="bg-white">
      <main>
        {/* Hero section */}
        <div className="overflow-hidden pt-8 sm:pt-12 lg:relative lg:py-48">
          <div className="mx-auto max-w-md px-6 sm:max-w-3xl lg:grid lg:max-w-7xl lg:grid-cols-2 lg:gap-24 lg:px-8">
            <div>
              <div>
                <img
                  className="h-32 w-auto"
                  src="logo-shape.png"
                  alt="Solenersync"
                />
              </div>
              <div className="mt-10">
                <div className="mt-6 sm:max-w-xl">
                  <h1 className="text-4xl font-bold tracking-tight text-gray-900 sm:text-5xl">
                    Solenersync
                  </h1>
                  <p className="mt-6 text-xl text-gray-500">
                    Synchronise your home energy with our solar irradiance
                    forecasting service
                  </p>
                </div>
                <form
                  onSubmit={handleSubmit}
                  className="mt-12 sm:flex sm:w-full sm:max-w-lg"
                >
                  <div className="min-w-0 flex-1">
                    <label htmlFor="hero-email" className="sr-only">
                      Email address
                    </label>
                    <input
                      onChange={(e) => setEmail(e.target.value)}
                      id="hero-email"
                      type="email"
                      className="block w-full rounded-md border border-gray-300 px-5 py-3 text-base text-gray-900 placeholder-gray-500 shadow-sm focus:border-rose-500 focus:ring-rose-500"
                      placeholder="Enter your email"
                    />
                  </div>
                  <div className="mt-4 sm:mt-0 sm:ml-3">
                    <button
                      type="submit"
                      className="block w-full rounded-md border border-transparent bg-rose-500 px-5 py-3 text-base font-medium text-white shadow hover:bg-rose-600 focus:outline-none focus:ring-2 focus:ring-rose-500 focus:ring-offset-2 sm:px-10"
                    >
                      Sign up
                    </button>
                  </div>
                </form>
                <div className="mt-4">
                  Already have an account?{" "}
                  <Link
                    className="text-rose-600 hover:text-rose-700"
                    href="/login"
                  >
                    Log in
                  </Link>
                </div>
                <div className="mt-10">
                  <div className="inline-flex items-center divide-x divide-gray-300">
                    <div className="flex flex-shrink-0 pr-5">
                      <StarIcon
                        className="h-5 w-5 text-yellow-400"
                        aria-hidden="true"
                      />
                      <StarIcon
                        className="h-5 w-5 text-yellow-400"
                        aria-hidden="true"
                      />
                      <StarIcon
                        className="h-5 w-5 text-yellow-400"
                        aria-hidden="true"
                      />
                      <StarIcon
                        className="h-5 w-5 text-yellow-400"
                        aria-hidden="true"
                      />
                      <StarIcon
                        className="h-5 w-5 text-yellow-400"
                        aria-hidden="true"
                      />
                    </div>
                    <div className="min-w-0 flex-1 py-1 pl-5 text-sm text-gray-500 sm:py-3">
                      <span className="font-medium text-gray-900">
                        Rated 5 stars
                      </span>{" "}
                      by over{" "}
                      <span className="font-medium text-rose-500">
                        500 beta users
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>

          <div className="sm:mx-auto sm:max-w-3xl sm:px-6">
            <div className="py-12 sm:relative sm:mt-12 sm:py-16 lg:absolute lg:inset-y-0 lg:right-0 lg:w-1/2">
              <div className="hidden sm:block">
                <div className="absolute inset-y-0 left-1/2 w-screen rounded-l-3xl bg-gray-50 lg:left-80 lg:right-0 lg:w-full" />
                <svg
                  className="absolute top-8 right-1/2 -mr-3 lg:left-0 lg:m-0"
                  width={404}
                  height={392}
                  fill="none"
                  viewBox="0 0 404 392"
                >
                  <defs>
                    <pattern
                      id="837c3e70-6c3a-44e6-8854-cc48c737b659"
                      x={0}
                      y={0}
                      width={20}
                      height={20}
                      patternUnits="userSpaceOnUse"
                    >
                      <rect
                        x={0}
                        y={0}
                        width={4}
                        height={4}
                        className="text-gray-200"
                        fill="currentColor"
                      />
                    </pattern>
                  </defs>
                  <rect
                    width={404}
                    height={392}
                    fill="url(#837c3e70-6c3a-44e6-8854-cc48c737b659)"
                  />
                </svg>
              </div>
              <div className="relative -mr-40 pl-6 sm:mx-auto sm:max-w-3xl sm:px-0 lg:h-full lg:max-w-none lg:pl-12">
                <img
                  className="w-full rounded-md shadow-xl ring-1 ring-black ring-opacity-5 lg:h-full lg:w-auto lg:max-w-none"
                  src="dashboard.png"
                  alt="dashboard"
                />
              </div>
            </div>
          </div>
        </div>
      </main>
      <Footer />
    </div>
  );
};

export default Home;
