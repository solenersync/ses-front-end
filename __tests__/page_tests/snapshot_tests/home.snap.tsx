import { render } from '@testing-library/react';
import Home from 'pages';
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

describe('Home', () => {
  const query = {
    email: 'jd@test.com',
  };

  (useRouter as jest.Mock).mockReturnValue({
    query: query,
  });

  test('should render the home page', async() => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
