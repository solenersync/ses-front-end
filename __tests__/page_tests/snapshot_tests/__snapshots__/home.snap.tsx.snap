// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the home page 1`] = `
<div>
  <div
    class="bg-white"
  >
    <main>
      <div
        class="overflow-hidden pt-8 sm:pt-12 lg:relative lg:py-48"
      >
        <div
          class="mx-auto max-w-md px-6 sm:max-w-3xl lg:grid lg:max-w-7xl lg:grid-cols-2 lg:gap-24 lg:px-8"
        >
          <div>
            <div>
              <img
                alt="Solenersync"
                class="h-32 w-auto"
                src="logo-shape.png"
              />
            </div>
            <div
              class="mt-10"
            >
              <div
                class="mt-6 sm:max-w-xl"
              >
                <h1
                  class="text-4xl font-bold tracking-tight text-gray-900 sm:text-5xl"
                >
                  Solenersync
                </h1>
                <p
                  class="mt-6 text-xl text-gray-500"
                >
                  Synchronise your home energy with our solar forecasting service
                </p>
              </div>
              <form
                class="mt-12 sm:flex sm:w-full sm:max-w-lg"
              >
                <div
                  class="min-w-0 flex-1"
                >
                  <label
                    class="sr-only"
                    for="hero-email"
                  >
                    Email address
                  </label>
                  <input
                    class="block w-full rounded-md border border-gray-300 px-5 py-3 text-base text-gray-900 placeholder-gray-500 shadow-sm focus:border-rose-500 focus:ring-rose-500"
                    id="hero-email"
                    placeholder="Enter your email"
                    type="email"
                  />
                </div>
                <div
                  class="mt-4 sm:mt-0 sm:ml-3"
                >
                  <button
                    class="block w-full rounded-md border border-transparent bg-rose-500 px-5 py-3 text-base font-medium text-white shadow hover:bg-rose-600 focus:outline-none focus:ring-2 focus:ring-rose-500 focus:ring-offset-2 sm:px-10"
                    type="submit"
                  >
                    Sign up
                  </button>
                </div>
              </form>
              <div
                class="mt-4"
              >
                Already have an account?
                 
                <a
                  class="text-rose-600 hover:text-rose-700"
                  href="/login"
                >
                  Log in
                </a>
              </div>
              <div
                class="mt-10"
              >
                <div
                  class="inline-flex items-center divide-x divide-gray-300"
                >
                  <div
                    class="flex flex-shrink-0 pr-5"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <div
                    class="min-w-0 flex-1 py-1 pl-5 text-sm text-gray-500 sm:py-3"
                  >
                    <span
                      class="font-medium text-gray-900"
                    >
                      Rated 5 stars
                    </span>
                     
                    by
                     
                    <span
                      class="font-medium text-rose-500"
                    >
                      beta users
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sm:mx-auto sm:max-w-3xl sm:px-6"
        >
          <div
            class="py-12 sm:relative sm:mt-12 sm:py-16 lg:absolute lg:inset-y-0 lg:right-0 lg:w-1/2"
          >
            <div
              class="hidden sm:block"
            >
              <div
                class="absolute inset-y-0 left-1/2 w-screen rounded-l-3xl bg-gray-50 lg:left-80 lg:right-0 lg:w-full"
              />
              <svg
                class="absolute top-8 right-1/2 -mr-3 lg:left-0 lg:m-0"
                fill="none"
                height="392"
                viewBox="0 0 404 392"
                width="404"
              >
                <defs>
                  <pattern
                    height="20"
                    id="837c3e70-6c3a-44e6-8854-cc48c737b659"
                    patternUnits="userSpaceOnUse"
                    width="20"
                    x="0"
                    y="0"
                  >
                    <rect
                      class="text-gray-200"
                      fill="currentColor"
                      height="4"
                      width="4"
                      x="0"
                      y="0"
                    />
                  </pattern>
                </defs>
                <rect
                  fill="url(#837c3e70-6c3a-44e6-8854-cc48c737b659)"
                  height="392"
                  width="404"
                />
              </svg>
            </div>
            <div
              class="relative -mr-40 pl-6 sm:mx-auto sm:max-w-3xl sm:px-0 lg:h-full lg:max-w-none lg:pl-12"
            >
              <img
                alt="dashboard"
                class="w-full rounded-md shadow-xl ring-1 ring-black ring-opacity-5 lg:h-full lg:w-auto lg:max-w-none"
                src="splash2.png"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="mt-24 sm:mt-6 text-center flex justify-center"
    >
      <div
        class="mx-auto max-w-md overflow-hidden py-12 px-6 sm:max-w-3xl lg:max-w-7xl lg:px-8"
      >
        <div
          class="mt-8 flex justify-center space-x-6"
        >
          <a
            href="https://www.solenersync.net/dashboard"
            rel="noopener noreferrer"
          >
            <img
              alt="Solenersync"
              class="md:h-20 h-16 w-auto"
              src="logo-small.png"
            />
          </a>
        </div>
        <p
          class="mt-8 text-center text-base text-gray-400"
        >
          Â© 2022 Solenersync. All rights reserved.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
